src/prefect/cli/flow_run.py

```diff
@@ -57,19 +57,24 @@ async def ls(
     flow_name: List[str] = typer.Option(None, help="Name of the flow"),
     limit: int = typer.Option(15, help="Maximum number of flow runs to list"),
     state: List[str] = typer.Option(None, help="Name of the flow run's state"),
-    state_type: List[StateType] = typer.Option(
-        None, help="Type of the flow run's state"
-    ),
+    state_type: List[str] = typer.Option(None, help="Type of the flow run's state"),
 ):
     """
     View recent flow runs or flow runs for specific flows
     """
+    for state_type_value in state_type:
+        enum_values = [item.value for item in StateType]
+        if state_type_value.upper() not in enum_values:
+            exit_with_error(
+                f"Invalid value for '--state-type': '{state_type_value}' is not one of"
+                f" '{enum_values}'."
+            )
```

Comment:
```suggestion
```

I don't think this is necessary. We should be able to continue to rely on the error handling from `typer`:
```bash
❯ prefect flow-run ls --state-type COMPLETED --state-type SADFSDFKJ
Usage: prefect flow-run ls [OPTIONS]
Try 'prefect flow-run ls --help' for help.
╭─ Error ──────────────────────────────────────────────────────────────────────╮
│ Invalid value for '--state-type': 'SADFSDFKJ' is not one of 'SCHEDULED',     │
│ 'PENDING', 'RUNNING', 'COMPLETED', 'FAILED', 'CANCELLED', 'CRASHED',         │
│ 'PAUSED', 'CANCELLING'.                                                      │
╰──────────────────────────────────────────────────────────────────────────────╯

```